sqlglot/parser.py

```diff
@@ -4484,6 +4492,46 @@ def _parse_as_command(self, start: Token) -> exp.Command:
         size = len(start.text)
         return exp.Command(this=text[:size], expression=text[size:])
 
+    def _parse_dict_property(self, this: str) -> exp.DictProperty:
+        settings = []
+
+        self._match_l_paren()
+        kind = self._parse_id_var()
+        if not kind:
+            self.raise_error(f"Unexpected token")
+        assert kind
```

Comment:
let's remove this assert and make 4519 kind.this more generous, this allows for partials

Reply:
I'm not sure about it as these are for dictionaries, and AFAIK dicts in CH do not work at all when the "kind" is unset.
See:
```
$ echo "create dictionary d (is UInt64, data String) SOURCE()" | clickhouse-format 
Code: 62. DB::Exception: Syntax error (query): failed at position 53 (')'): )
. Expected one of: function with key-value arguments, identifier. (SYNTAX_ERROR), Stack trace (when copying this message, always include the lines below):

0. ./build_docker/./src/Common/Exception.cpp:92: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000dcaeaf5 in /usr/bin/clickhouse
1. DB::Exception::createDeprecated(String const&, int, bool) @ 0x0000000008ace3ed in /usr/bin/clickhouse
2. ./build_docker/./src/Parsers/parseQuery.cpp:362: DB::parseQueryAndMovePosition(DB::IParser&, char const*&, char const*, String const&, bool, unsigned long, unsigned long) @ 0x0000000014c64300 in /usr/bin/clickhouse
3. ./build_docker/./programs/format/Format.cpp:163: mainEntryClickHouseFormat(int, char**) @ 0x000000000de998ce in /usr/bin/clickhouse
4. ./build_docker/./programs/main.cpp:0: main @ 0x000000000874616f in /usr/bin/clickhouse
5. ? @ 0x00007fc82b329d90 in ?
6. __libc_start_main @ 0x00007fc82b329e40 in ?
7. _start @ 0x0000000007e8d3ae in /usr/bin/clickhouse
 (version 23.5.1.1345 (official build))
```

Reply:
Note: `clickhouse-format` is not a validator, it only checks the syntax. So I would like `sqlglot` to have exactly the same behavior.

Reply:
it doesn't matter, as i said this is for partials. and sqlglot is not a validator

we want to be able to parse with error mode ignore

create dictionary d(...) SOURCE()

Reply:
ah, ok